projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,327 @@
+    private boolean sentHaser = true;                 // Indicates whether user has sent.
+    private boolean sentMoreThanFiveLawsuits = false; // Indicates whether user has sent more then 5 lawsuits already in the past year.
+    private String moreThanFiveLawsuits ="הגיש";
+    private String lessThanFiveLawsuits = "לא הגיש/ו";
+    private String fiveLawsuitsStatus ="";            // Init onCreate().
+    private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    private String claimCaseSubscription = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    private String claimCase = "";                    // "Didn't subscribe" / "Sent 'haser' but got spam". Set on fillInTemplate().
+
+    // User:
+    private String userPrivateName = "שם פרטי";
+    private String userLastname = "שם משפחה";
+    private String userID = "ת ז משתמש";
+    private String userAddress = "כתובת משתמש";
+    private String userPhone = "טלפון משתמש";
+    private String userFax = "פקס משתמש";
+
+    // First spam company's details:
+    private String companyName = "שם החברה";
+    private String companyId = "מספר ח.פ";
+    private String companyAddress = "כתובת החברה";
+    private String companyPhone = "טלפון החברה";
+    private String companyFax = "פקס חברה";
+
+    // Second spam company's det ails:
+    private String company2Name = "שם החברה 2";
+    private String company2Id = "מספר ח.פ 2";
+    private String company2Address = "כתובת החברה 2";
+    private String company2Phone = "טלפון החברה 2";
+    private String company2Fax = "פקס חברה 2";
+// ----------------------------- - -------------------------------------------------------:
+    Button button_create_pdf;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_lawsuit_pdf);
+        button_create_pdf = (Button) findViewById(R.id.button_createPDF);
+
+        // Check Read/Write permissions:
+        isWriteStoragePermissionGranted();
+
+        // Initialize folders needed for activity:
+        initDirStructure();
+
+        // Read lawsuit template's content:
+        templateContent = readTemplate();
+
+        // Init dates:
+        receivedSpamDate = new Date();
+        currentDate = new Date();
+        // Update date values:
+        dateAndTimeFormatter = new SimpleDateFormat(DATE_TIME_FORMAT);
+        dateAndTimeFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+        dateFormatter = new SimpleDateFormat(DATE_FORMAT);
+        dateFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+
+        // Generate lawsuit PDF OnClick:
+        button_create_pdf.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    createPdfLawsuitFileName();
+                    createPdf();
```

Comment:
It seems to me that, after your changes to this PR, `createPdfLawsuitFileName()` will just set `lawsuitOutputPdfFilePath`, so you can simplify the code by not having such a member, and just calling the method straight from `createPdf()` into a local method variable inside `createPdf()`.

Reply:
The thought was to have the date as close it can be to when user confirmed the form.
I changed createPdfLawsuitFileName() to be responsible just to generate the file name. The path for it is now generated inside the createPdf().